File: library-base/src/main/java/com/my/library_base/base/BaseViewModel.java
package com.my.library_base.base;

import android.app.Application;


import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleOwner;

import com.my.library_base.base.inf.IBaseViewModel;


import io.reactivex.disposables.Disposable;

public class BaseViewModel<M extends BaseModel> extends AndroidViewModel implements IBaseViewModel {
    protected M model;

    public BaseViewModel(@NonNull Application application) {
        super(application);
    }

    public BaseViewModel(@NonNull Application application, M model) {
        super(application);
        this.model = model;
    }

    protected void addSubscribe(Disposable disposable) {
        model.addSubscribe(disposable);
    }

    public M getModel() {
        return model;
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        if (model != null) {
            model.onCleared();
        }
    }


    @Override
    public void onAny(LifecycleOwner owner, Lifecycle.Event event) {

    }

    @Override
    public void onCreate() {

    }

    @Override
    public void onDestroy() {

    }

    @Override
    public void onStart() {

    }

    @Override
    public void onStop() {

    }

    @Override
    public void onResume() {

    }

    @Override
    public void onPause() {

    }

    @Override
    public void registerRxBus() {

    }

    @Override
    public void removeRxBus() {

    }
}

File: library-base/src/main/java/com/my/library_base/base/BaseViewModel.kt
package com.my.library_base.base

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleOwner
import com.my.library_base.base.inf.IBaseViewModel
import io.reactivex.disposables.Disposable

open class BaseViewModel<M : BaseModel?> : AndroidViewModel, IBaseViewModel {
    var model: M? = null
        protected set

    constructor(application: Application) : super(application) {}
    constructor(application: Application, model: M) : super(application) {
        this.model = model
    }

    protected fun addSubscribe(disposable: Disposable?) {
        model!!.addSubscribe(disposable)
    }

    override fun onCleared() {
        super.onCleared()
        model?.onCleared()
    }

    override fun onAny(owner: LifecycleOwner?, event: Lifecycle.Event?) {}
    override fun onCreate() {}
    override fun onDestroy() {}
    override fun onStart() {}
    override fun onStop() {}
    override fun onResume() {}
    override fun onPause() {}
    override fun registerRxBus() {}
    override fun removeRxBus() {}
}